### DISCUSSION TITLE
RectangleTexture - prevent rounded corners from skewing with aspect ratio

### DISCUSSION THREAD

**Discussion by mariomatasystems** [cables-gl/cables_docs#908] [Feature Requests] [answered]
I notice if the RectangleTexture is not 1:1 the rounded corners skew,

<img width="507" alt="Screenshot 2024-11-12 at 21 55 28" src="https://github.com/user-attachments/assets/0b12581f-3a92-4554-ba85-f2a6cf8fe97b">

It would be more useful to keep the rounded corners consistent

Might be wrong (chatgpt) but i think we can tweak the frag code to account for the aspect.

changing this

    float roundn=roundness*min(size.x,size.y);

to this

    vec2 aspectScale = vec2(1.0, size.y / size.x);
    float roundn = roundness * length(size * aspectScale);

**Comment by pandrr** [comment]
thanks, good input!

**Comment by pandrr** [comment, marked as answer]
created issue: https://github.com/cables-gl/cables_docs/issues/909
